components/audio-context-provider.tsx

```diff
@@ -0,0 +1,33 @@
+'use client';
+import {useEffect, useState} from 'react';
+import {Context} from '../contexts/audio';
+
+export const AudioContextProvider = ({children}: {children: React.ReactNode}): JSX.Element => {
+	const [audioContext, setAudioContext] = useState<AudioContext>();
+
+	useEffect(() => {
+		// Create and set up the AudioContext when the component mounts
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		const context = new AudioContext();
+		setAudioContext(context);
+
+		// Clean up the AudioContext when the component unmounts
+		return () => {
+			context.close().catch((error) => {
+				console.error('Error closing AudioContext:', error);
+			});
+		};
+	}, []);
+
+	useEffect(() => {
+		console.log('new audioContext', audioContext);
+	}, [audioContext]);
+
+	return (
+		<Context.Provider value={audioContext}>
+			{children}
+		</Context.Provider>
+	);
+};
+
```

Comment:
The audio context provider is unnecessary; nothing else across the app requires access to this context apart from the audio hook. Therefore, the creation of `AudioContext` should be co-located in the audio hook. Unless I'm missing something critical here, you can explain your thinking.

Reply:
Created it there to abstract away the audioContext since the useAudio hook was quite specific to events only, and it doesn't really make that much sense from what i gather to have multiple audiocontexts around if you wanted sound for something else down the line. But I agree co-location for now is much cleaner